pkg/operator/controller/certificate/ca.go

```diff
@@ -0,0 +1,152 @@
+package certificate
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// caCertSecretName is the name of the secret that holds the CA certificate
+	// that the operator will use to create default certificates for
+	// clusteringresses.
+	caCertSecretName = "router-ca"
+)
+
+// CASecretName returns the namespaced name for the router CA secret.
+func CASecretName(operatorNamespace string) types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: operatorNamespace,
+		Name:      caCertSecretName,
+	}
+}
+
+func (r *reconciler) ensureRouterCASecret() (*corev1.Secret, error) {
+	current, err := r.currentRouterCASecret()
+	if err != nil {
+		return nil, err
+	}
+	if current != nil {
+		return current, nil
+	}
+	desired, err := desiredRouterCASecret(r.operatorNamespace)
```

Comment:
`desiredRouterCASecret` will generate new cert every time and current != desired after first iteration.
May be we could just use get + create resource pattern that we use in other places to avoid confusion? or we could rename `desired` to something more accurate.

Reply:
We're not currently diffing them — are you saying there's an error here, or just that it's confusing?

If desiredRouterCASecret isn't deterministic that seems like a problem, and if it _can't be_ deterministic then our diffing logic would have to account for it (if/when we start diffing)

Reply:
`routerCAConfigMapsEqual` compares only `.data`, similar to how `deploymentConfigChanged` compares only select fields.  I think this pattern is reasonable since there will always be fields that should be ignored (e.g., uid).

Reply:
There is no error here. It was confusing for me when i saw current can be different from desired (issue, expiry times will differ as those are derived from current time). Ignore if you think this does not confuse others.

Reply:
Agree it's weird, and we can fix it in a followup as it's no different than today — I just transferred the logic over as close to the same as possible for the restructuring